packages/api/src/repository.js

```diff
@@ -50,28 +49,22 @@ class Repository {
     return (querySnapshot.size === 1);
   }
 
-  async sessionPermissions (sessionID) {
-    const docRef = client.doc('express-sessions/' + sessionID);
-    const solution = { permitted: false, expiration: null, login: null };
-    const doc = await docRef.get();
-    let data;
-    try {
-      data = await JSON.parse(doc.data().data);
-    } catch (err) {
-      return solution;
-    }
-    if (doc.exists) {
-      const expiration = data.expires;
-      if (expiration == null) {
-        return solution;
-      }
-      if (moment().isBefore(moment(expiration))) {
-        solution.permitted = true;
-        solution.expiration = moment(expiration).format();
-        solution.login = data.user;
-      }
-    }
-    return solution;
+  async storeTicket (ticketToPerform) {
+    await this.createDoc('tickets/ticket-to-perform', {
```

Comment:
I believe this will keep overwriting a document named `tickets/ticket-to-perform`. Instead of doing this, I would just use the same unique identifier you've already created for `state` and create a document:

```
ticets/${state}
```

I think we should consider getting all this database work hammered out in a second PR after we land this though.